sycl-fusion/passes/target/TargetFusionInfo.h

```diff
@@ -0,0 +1,186 @@
+//==-- TargetFusionInfo.h - Encapsule target-specific fusion functionality -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SYCL_FUSION_PASSES_TARGETFUSIONINFO_H
+#define SYCL_FUSION_PASSES_TARGETFUSIONINFO_H
+
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Module.h"
+
+namespace llvm {
+
+class TargetFusionInfoImpl {
```

Comment:
To better use the PIMPL idiom, I propose to move implementation definitions to the `.cpp` file, having `TargetFusionInfoImpl` as a private member type of `TargetFusionInfo`, making the impl pointer `unique_ptr` and not passing the implementation to the `TargetFusionInfo` constructor.